confgenerator/otel/processors.go

```diff
@@ -47,6 +48,18 @@ func MetricsTransform(metrics ...map[string]interface{}) Component {
 	}
 }
 
+// AddPrefix returns a config snippet that adds a prefix to all metrics.
+func AddPrefix(prefix string) map[string]interface{} {
+	// $ needs to be escaped because reasons.
+	// https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/metricstransformprocessor#rename-multiple-metrics-using-substitution
+	return map[string]interface{}{
+		"include":    `^(.*)$$`,
+		"match_type": "regexp",
+		"action":     "update",
+		"new_name":   path.Join(prefix, `$${1}`),
```

Comment:
Erm, wouldn't this use the `\\` separator when run on Windows? Why not just use ``prefix+`/$${1}` ``? Or, better yet, just set `prefix` to `"agent.googleapis.com/"` (with the trailing slash), to match the semantics of the exporter…

Reply:
These are URLs, does windows really use \\ for urls?

Reply:
No, `path` always uses forward slash. It's specifically designed for the URL usecase. (Also it's what the OT code itself uses.)

`filepath` is the package that would use `\` on Windows.

You are incorrect about the semantics of the exporter; the trailing slash is not necessary because the exporter also uses `path.Join`.